// @aws-sdk/client-sns/dist-cjs/models/errors.js
const { SNSServiceException: __BaseException } = require("./SNSServiceException");
exports.AuthorizationErrorException = class AuthorizationErrorException extends __BaseException {
    name = "AuthorizationErrorException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AuthorizationErrorException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AuthorizationErrorException.prototype);
    }
};
exports.InternalErrorException = class InternalErrorException extends __BaseException {
    name = "InternalErrorException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "InternalErrorException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalErrorException.prototype);
    }
};
exports.InvalidParameterException = class InvalidParameterException extends __BaseException {
    name = "InvalidParameterException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidParameterException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidParameterException.prototype);
    }
};
exports.NotFoundException = class NotFoundException extends __BaseException {
    name = "NotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NotFoundException.prototype);
    }
};
exports.ThrottledException = class ThrottledException extends __BaseException {
    name = "ThrottledException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ThrottledException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottledException.prototype);
    }
};
exports.FilterPolicyLimitExceededException = class FilterPolicyLimitExceededException extends __BaseException {
    name = "FilterPolicyLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "FilterPolicyLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, FilterPolicyLimitExceededException.prototype);
    }
};
exports.ReplayLimitExceededException = class ReplayLimitExceededException extends __BaseException {
    name = "ReplayLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ReplayLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ReplayLimitExceededException.prototype);
    }
};
exports.SubscriptionLimitExceededException = class SubscriptionLimitExceededException extends __BaseException {
    name = "SubscriptionLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "SubscriptionLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, SubscriptionLimitExceededException.prototype);
    }
};
exports.OptedOutException = class OptedOutException extends __BaseException {
    name = "OptedOutException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "OptedOutException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OptedOutException.prototype);
    }
};
exports.UserErrorException = class UserErrorException extends __BaseException {
    name = "UserErrorException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "UserErrorException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UserErrorException.prototype);
    }
};
exports.ConcurrentAccessException = class ConcurrentAccessException extends __BaseException {
    name = "ConcurrentAccessException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConcurrentAccessException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConcurrentAccessException.prototype);
    }
};
exports.InvalidSecurityException = class InvalidSecurityException extends __BaseException {
    name = "InvalidSecurityException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidSecurityException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidSecurityException.prototype);
    }
};
exports.StaleTagException = class StaleTagException extends __BaseException {
    name = "StaleTagException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "StaleTagException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StaleTagException.prototype);
    }
};
exports.TagLimitExceededException = class TagLimitExceededException extends __BaseException {
    name = "TagLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TagLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TagLimitExceededException.prototype);
    }
};
exports.TagPolicyException = class TagPolicyException extends __BaseException {
    name = "TagPolicyException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TagPolicyException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TagPolicyException.prototype);
    }
};
exports.TopicLimitExceededException = class TopicLimitExceededException extends __BaseException {
    name = "TopicLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TopicLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TopicLimitExceededException.prototype);
    }
};
exports.ResourceNotFoundException = class ResourceNotFoundException extends __BaseException {
    name = "ResourceNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
    }
};
exports.InvalidStateException = class InvalidStateException extends __BaseException {
    name = "InvalidStateException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidStateException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidStateException.prototype);
    }
};
exports.ValidationException = class ValidationException extends __BaseException {
    name = "ValidationException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.Message = opts.Message;
    }
};
exports.EndpointDisabledException = class EndpointDisabledException extends __BaseException {
    name = "EndpointDisabledException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "EndpointDisabledException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, EndpointDisabledException.prototype);
    }
};
exports.InvalidParameterValueException = class InvalidParameterValueException extends __BaseException {
    name = "InvalidParameterValueException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidParameterValueException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidParameterValueException.prototype);
    }
};
exports.KMSAccessDeniedException = class KMSAccessDeniedException extends __BaseException {
    name = "KMSAccessDeniedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KMSAccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KMSAccessDeniedException.prototype);
    }
};
exports.KMSDisabledException = class KMSDisabledException extends __BaseException {
    name = "KMSDisabledException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KMSDisabledException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KMSDisabledException.prototype);
    }
};
exports.KMSInvalidStateException = class KMSInvalidStateException extends __BaseException {
    name = "KMSInvalidStateException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KMSInvalidStateException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KMSInvalidStateException.prototype);
    }
};
exports.KMSNotFoundException = class KMSNotFoundException extends __BaseException {
    name = "KMSNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KMSNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KMSNotFoundException.prototype);
    }
};
exports.KMSOptInRequired = class KMSOptInRequired extends __BaseException {
    name = "KMSOptInRequired";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KMSOptInRequired",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KMSOptInRequired.prototype);
    }
};
exports.KMSThrottlingException = class KMSThrottlingException extends __BaseException {
    name = "KMSThrottlingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KMSThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KMSThrottlingException.prototype);
    }
};
exports.PlatformApplicationDisabledException = class PlatformApplicationDisabledException extends __BaseException {
    name = "PlatformApplicationDisabledException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "PlatformApplicationDisabledException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PlatformApplicationDisabledException.prototype);
    }
};
exports.BatchEntryIdsNotDistinctException = class BatchEntryIdsNotDistinctException extends __BaseException {
    name = "BatchEntryIdsNotDistinctException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "BatchEntryIdsNotDistinctException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BatchEntryIdsNotDistinctException.prototype);
    }
};
exports.BatchRequestTooLongException = class BatchRequestTooLongException extends __BaseException {
    name = "BatchRequestTooLongException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "BatchRequestTooLongException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BatchRequestTooLongException.prototype);
    }
};
exports.EmptyBatchRequestException = class EmptyBatchRequestException extends __BaseException {
    name = "EmptyBatchRequestException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "EmptyBatchRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, EmptyBatchRequestException.prototype);
    }
};
exports.InvalidBatchEntryIdException = class InvalidBatchEntryIdException extends __BaseException {
    name = "InvalidBatchEntryIdException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidBatchEntryIdException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidBatchEntryIdException.prototype);
    }
};
exports.TooManyEntriesInBatchRequestException = class TooManyEntriesInBatchRequestException extends __BaseException {
    name = "TooManyEntriesInBatchRequestException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyEntriesInBatchRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyEntriesInBatchRequestException.prototype);
    }
};
exports.VerificationException = class VerificationException extends __BaseException {
    name = "VerificationException";
    $fault = "client";
    Message;
    Status;
    constructor(opts) {
        super({
            name: "VerificationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, VerificationException.prototype);
        this.Message = opts.Message;
        this.Status = opts.Status;
    }
};
